src/lib/selective-sync.js

```diff
@@ -0,0 +1,52 @@
+var debug = require('debug')('dat')
+var path = require('path')
+var EventEmitter = require('events').EventEmitter
+
+module.exports = function selectiveSync (state, bus) {
+  var archive = state.dat.archive
+  debug('sparse mode. downloading metadata')
+  var emitter = new EventEmitter()
+
+  function download (entry) {
+    debug('selected', entry)
+    archive.stat(entry, function (err, stat) {
+      if (err) return bus.emit('exit:error', err)
```

Comment:
I'd be nice if we can warn without exiting if they select a file that doesn't exist. But that may be a harder UI problem we can solve later.